src/dbos.ts

```diff
@@ -0,0 +1,893 @@
+import { Span } from "@opentelemetry/sdk-trace-base";
+import {
+  assertCurrentDBOSContext,
+  assertCurrentWorkflowContext,
+  getCurrentContextStore,
+  getCurrentDBOSContext,
+  HTTPRequest,
+  runWithTopContext,
+  DBOSContextImpl
+} from "./context";
+import { DBOSConfig, DBOSExecutor, InternalWorkflowParams } from "./dbos-executor";
+import {
+  GetWorkflowQueueInput,
+  GetWorkflowQueueOutput,
+  GetWorkflowsInput,
+  GetWorkflowsOutput,
+  WorkflowConfig,
+  WorkflowFunction,
+  WorkflowParams
+} from "./workflow";
+import { DBOSExecutorContext } from "./eventreceiver";
+import { DLogger, GlobalLogger } from "./telemetry/logs";
+import { DBOSExecutorNotInitializedError, DBOSInvalidWorkflowTransitionError } from "./error";
+import { parseConfigFile } from "./dbos-runtime/config";
+import { DBOSRuntimeConfig } from "./dbos-runtime/runtime";
+import { DBOSScheduler, ScheduledArgs, SchedulerConfig, SchedulerRegistrationBase } from "./scheduler/scheduler";
+import { configureInstance, getOrCreateClassRegistration, getRegisteredOperations, MethodRegistration, registerAndWrapContextFreeFunction, registerFunctionWrapper } from "./decorators";
+import { sleepms } from "./utils";
+import { DBOSHttpServer } from "./httpServer/server";
+import { koaTracingMiddleware, expressTracingMiddleware } from "./httpServer/middleware";
+import { Server } from "http";
+import { DrizzleClient, PrismaClient, TypeORMEntityManager, UserDatabaseClient } from "./user_database";
+import { TransactionConfig, TransactionContextImpl, TransactionFunction } from "./transaction";
+
+import Koa from "koa";
+import { Application as ExpressApp } from "express";
+import { INestApplication } from "@nestjs/common";
+import { FastifyInstance } from "fastify";
+import _fastifyExpress from "@fastify/express"; // This is for fastify.use()
+
+import { PoolClient } from "pg";
+import { Knex } from "knex";
+import { StepConfig, StepFunction } from "./step";
+import { wfQueueRunner } from "./wfqueue";
+import {
+  HandlerContext,
+  StepContext,
+  StoredProcedureContext,
+  TransactionContext,
+  WorkflowContext,
+  WorkflowHandle
+} from ".";
+import { ConfiguredInstance } from ".";
+import { StoredProcedureFunc } from "./procedure";
+import { APITypes } from "./httpServer/handlerTypes";
+import { HandlerRegistrationBase } from "./httpServer/handler";
+
+// Declare all the HTTP applications a user can pass to the DBOS object during launch()
+// This allows us to add a DBOS tracing middleware (extract W3C Trace context, set request ID, etc)
+export interface DBOSHttpApps {
```

Comment:
Hmm, what if there are multiple apps? Do we allow multiple of those (e.g., both koaApp and expressApp)?

Reply:
@maxdml 
You probably should just do 1, or extract the routes and make a single handler tree, I don't think you'd have a lot of luck getting multiple ports through the LB right now.